### PR TITLE
Group math package functions

### PR DESCRIPTION
**Pull request by janekdb**
Group math package functions

Groups
- Mathematical Constants
- Minimum and Maximum
- Rounding
- Exponential and Logarithmic
- Trigonometric
- Angular Measurement Conversion
- Hyperbolic
- Absolute Values
- Signs
- Root Extraction
- Polar Coordindates
- Unit of Least Precision

Other changes,
- Dropped use of `double` for `Double` in all cases
- Grouped some methods in the source
- Extended notes about exception to method forwarding
- Minor method documentation enhancements

IEEERemainder is in the Rounding group since it is related to rounding and did not justify a new group.
## Review Snapshot

The Scaladoc is published here for review purposes: http://janekdb.github.io/scala/PR-5225/scala/math/index.html

### CODE REVIEW

**Comment by janekdb** [comment]
I have added [SI-9813](https://issues.scala-lang.org/browse/SI-9813) to cover the addition of the missing Java 1.6 and 1.8 methods. Happy to pick this up following agreement it should be done.

**Comment by janekdb** [comment]
Review by anyone motivated sufficiently to do so!

File: src/library/scala/Predef.scala [hunk 1]
```diff
@@ -71,7 +71,7 @@ import scala.io.StdIn
  *
  * @groupname assertions Assertions
  * @groupprio assertions 20
- * @groupdesc assertions These methods support program verfication and runtime correctness.
+ * @groupdesc assertions These methods support program verification and runtime correctness.
```
Comment (som-snytt) [inline src/library/scala/Predef.scala:74]: Are you sure it wasn't supposed to be `nerfication`?

Comment (janekdb) [inline src/library/scala/Predef.scala:74]: Absolutely sure it should be `vitrification`.

**Comment by janekdb** [comment]
Review by @martijnhoekstra . Any comments welcome.

**Comment by martijnhoekstra** [comment]
hi @janekdb , I'm in really no position to review this (or pretty much anything for that matter). That said, looks like a nice improvement to me. I assume things are a bit quiet at the PR review queue front due to just coming out of crunch mode for 2.12.0-M5

**Comment by janekdb** [comment]
Thanks @martijnhoekstra  !

File: src/library/scala/math/package.scala [hunk 2]
```diff
@@ -11,35 +11,98 @@ package scala
...
+  * Note that these are not pure forwarders to the java versions.
+  * In particular, the return type of java.lang.Long.signum is Int,
+  * but here it is widened to Long so that each overloaded variant
+  * will return the same numeric type it is passed.
+  * @groupprio signum 90
+  *
+  * @groupname root-extraction Root Extraction
+  * @groupprio root-extraction 100
+  *
+  * @groupname polar-coords Polar Coordindates
```
Comment (adriaanm) [inline src/library/scala/math/package.scala]: As in "Waiter! There's a cord in my dates!?" 😎

File: src/library/scala/math/package.scala [hunk 3]
```diff
@@ -11,35 +11,98 @@ package scala
...
+  * @groupprio polar-coords 110
+  *
+  * @groupname ulp Unit of Least Precision
+  * @groupprio ulp 120
+  *
+  * @groupname randomisation Psuedo Random Number Generation
```
Comment (adriaanm) [inline src/library/scala/math/package.scala]: `s/ue/eu/`

**Comment by adriaanm** [comment]
Very nice! Caught a few typos, but otherwise looks good to go!

File: src/library/scala/math/package.scala (hunk 2 above)
Comment (janekdb) [inline src/library/scala/math/package.scala]: Fixed!

File: src/library/scala/math/package.scala (hunk 3 above)
Comment (janekdb) [inline src/library/scala/math/package.scala]: Fixed!

**Comment by janekdb** [comment]
Fixes applied, rebased and pushed.

**Comment by janekdb** [comment]
Build failure

```
OpenJDK 64-Bit Server VM warning:
INFO: os::commit_memory(0x00000007a7900000, 80740352, 0) failed;
error='Cannot allocate memory' (errno=12)
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (mmap) failed to map 80740352 bytes for committing reserved memory.
# An error report file with more information is saved as:
# /home/jenkins/workspace/scala-2.12.x-validate-test/hs_err_pid28480.log
```

**Comment by janekdb** [comment]
@adriaanm - Typos fixed, build ok. Please rereview.

File: src/library/scala/math/package.scala [hunk 4]
```diff
@@ -48,155 +111,217 @@ package object math {
...
-   *  In particular, the return type of java.lang.Long.signum is Int,
-   *  but here it is widened to Long so that each overloaded variant
-   *  will return the same numeric type it is passed.
-   */
+  /** @group signum
+    * @note Forwards to [[java.lang.Integer]]
+    */
   def signum(x: Int): Int       = java.lang.Integer.signum(x)
+  /** @group signum
+    * @note Forwards to [[java.lang.Long]]
```
Comment (som-snytt) [inline src/library/scala/math/package.scala:214]: Intended to omit note that return is widened? Or is that too banal?

Comment (som-snytt) [inline src/library/scala/math/package.scala:214]: nm, I'd forgotten I'd seen it at top...

**Comment by som-snytt** [comment]
LGTM

**Pull request merged**